iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -21,11 +21,20 @@
 #include "iceoryx_posh/internal/runtime/posh_runtime_impl.hpp"
 
 #include <cstdint>
+#include <new>
+#include <type_traits>
 
 namespace iox
 {
 namespace runtime
 {
+
+// A refcount for use in getLifetimeParticipant().
+static int refcount;
```

Comment:
We always use types with explicit sizes. Could you please use `uint64_t` here and initialize the `refcount` and rename it into `s_refcount`.
```
static uint64_t s_refcount{0U};
```

Reply:
This must be a `std::atomic` since `getLifetimeParticipant` could be called from any thread. In addition, we might have the [static initialization fiasco](https://www.cs.technion.ac.il/users/yechiel/c++-faq/static-init-order.html), which is the reason this mechanism is so cumbersome. It just occurred to me that I might have introduced this fiasco by myself lately :grimacing:

Reply:
Yesterday after work I also realized that these things should be atomic :)

Reply:
Happens also way to often with me in C++. It's just too easy to make it wrong. We should RiiR :grin: